opensquirrel/decomposer/mckay_decomposer.py

```diff
@@ -30,8 +31,20 @@ def decompose(self, g: Gate) -> list[Gate]:
         if g.name == "Rz" or g.name == "X90":
             return [g]
 
-        # McKay decomposition
+        if g.axis[0] == 0 and g.axis[1] == 0:
+            rz_angle = float(g.angle * g.axis[2])
+            return [Rz(g.get_qubit_operands()[0], Float(rz_angle))]
+
+        zxz_decomposition = ZXZDecomposer().decompose(g)
+        zxz_angle = 0.0
+        if len(zxz_decomposition) >= 2 and isinstance(zxz_decomposition[1], BlochSphereRotation):
+            zxz_angle = zxz_decomposition[1].angle
+
+        if zxz_angle == pi / 2:
+            zxz_decomposition[1] = X90(g.get_qubit_operands()[0])
```

Comment:
Why is `rz_angle = float(g.angle * g.axis[2])`; I understand the angle, but not why it needs to be multiplied by the z-component of the axis...

```suggestion
            rz_angle = float(g.angle * g.axis[2])
            return [Rz(g.qubit, Float(rz_angle))]

        zxz_decomposition = ZXZDecomposer().decompose(g)
        zxz_angle = 0.0
        if len(zxz_decomposition) >= 2 and isinstance(zxz_decomposition[1], BlochSphereRotation):
            zxz_angle = zxz_decomposition[1].angle

        if abs(zxz_angle - pi / 2) < ATOL:
            zxz_decomposition[1] = X90(g.qubit)
```